br/pkg/gluetidb/glue.go

```diff
@@ -203,13 +203,33 @@ func (gs *tidbSession) CreatePlacementPolicy(ctx context.Context, policy *model.
 	return d.CreatePlacementPolicyWithInfo(gs.se, policy, ddl.OnExistIgnore)
 }
 
+// SplitBatchCreateTable provide a way to split batch into small batch when batch size is large than 6 MB.
+// The raft entry has limit size of 6 MB, a batch of CreateTables may hit this limitation
+// TODO: shall query string be set for each split batch create, it looks does not matter if we set once for all.
+func (gs *tidbSession) SplitBatchCreateTable(schema model.CIStr, info []*model.TableInfo, l int, r int, cs ...ddl.CreateTableWithInfoConfigurier) error {
```

Comment:
In fact we can make subranges of slices fast, perhaps consider remove the `l` and `r` arguments, and:
```go
if err := BatchCreateTable(info); isEntryTooLarge(err) {
  if len(info) == 1 { return err }
  mid := len(info) / 2
  if err := BatchCreateTable(info[:mid]); err != nil { return err }
  if err := BatchCreateTable(info[mid:]); err != nil { return err }
  return nil
}
```

Reply:
fix in [2e0df43](https://github.com/pingcap/tidb/pull/39829/commits/2e0df43482cc4bd06cabe36cbcec643dd5840709)